File: medium/1500-1999/1679-max-number-of-k-sum-pairs/sol2.cpp
#include <bits/stdc++.h>
using namespace std;

class Solution
{
public:
    int maxOperations(vector<int> &nums, int k)
    {
        int ans = 0, left = 0, right = nums.size() - 1;

        sort(nums.begin(), nums.end());

        while (left < right)
        {
            if (nums[left] + nums[right] == k)
                ans++, left++, right--;
            else if (nums[left] + nums[right] < k)
                left++;
            else
                right--;
        }

        return ans;
    }
};

int main()
{
    cout << "Hello C++" << endl;
    return 0;
}

File: medium/2000-2499/2130-maximum-twin-sum-of-a-linked-list/sol2.py
from typing import Optional

# Definition for singly-linked list.
class ListNode:
    def __init__(self, val=0, next=None):
        self.val = val
        self.next = next

# O(n) time complexity, O(1) space complexity
class Solution:
    def pairSum(self, head: Optional[ListNode]) -> int:
        slow, fast = head, head
        while fast and fast.next:
            slow = slow.next
            fast = fast.next.next
        
        # reverse 2nd half
        prev, curr = None, slow
        while curr:
            temp = curr.next
            curr.next = prev
            
            prev = curr
            curr = temp
        
        ans = 0
        left, right = head, prev
        while right:
            ans = max(ans, left.val + right.val)
            left = left.next
            right = right.next
        
        return ans